File: frontend/docs/examples/python/custom_message_publisher.py
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import JointState
from std_msgs.msg import Header
from rclpy.qos import QoSProfile, ReliabilityPolicy, DurabilityPolicy


class CustomMessagePublisher(Node):

    def __init__(self):
        super().__init__('custom_message_publisher')

        # Create publisher with custom QoS settings
        qos_profile = QoSProfile(
            depth=10,
            reliability=ReliabilityPolicy.BEST_EFFORT,
            durability=DurabilityPolicy.VOLATILE
        )

        self.publisher = self.create_publisher(JointState, 'humanoid_joints', qos_profile)
        self.timer = self.create_timer(0.1, self.publish_joint_states)  # 10 Hz

    def publish_joint_states(self):
        msg = JointState()
        msg.name = ['hip_left', 'knee_left', 'ankle_left', 'hip_right', 'knee_right', 'ankle_right']
        msg.position = [0.1, 0.2, 0.3, 0.1, 0.2, 0.3]
        msg.velocity = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
        msg.effort = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
        msg.header.stamp = self.get_clock().now().to_msg()
        msg.header.frame_id = 'base_link'

        self.publisher.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = CustomMessagePublisher()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: i18n/en/docusaurus-plugin-content-docs/current/examples/cpp/custom_message_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <sensor_msgs/msg/joint_state.hpp>

class CustomMessagePublisher : public rclcpp::Node
{
public:
    CustomMessagePublisher()
    : Node("custom_message_publisher")
    {
        // Create publisher with custom QoS settings
        rclcpp::QoS qos_profile(10);
        qos_profile.reliability(RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT);
        qos_profile.durability(RMW_QOS_POLICY_DURABILITY_VOLATILE);

        publisher_ = this->create_publisher<sensor_msgs::msg::JointState>(
            "humanoid_joints", qos_profile);

        timer_ = this->create_wall_timer(
            std::chrono::milliseconds(100),
            std::bind(&CustomMessagePublisher::publish_joint_states, this));
    }

private:
    void publish_joint_states()
    {
        auto msg = sensor_msgs::msg::JointState();
        msg.name = {"hip_left", "knee_left", "ankle_left", "hip_right", "knee_right", "ankle_right"};
        msg.position = {0.1, 0.2, 0.3, 0.1, 0.2, 0.3};
        msg.velocity = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
        msg.effort = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
        msg.header.stamp = this->now();
        msg.header.frame_id = "base_link";

        publisher_->publish(msg);
    }

    rclcpp::Publisher<sensor_msgs::msg::JointState>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<CustomMessagePublisher>());
    rclcpp::shutdown();
    return 0;
}